packages/apollo/src/generate.ts

```diff
@@ -156,25 +156,27 @@ export default function generate(
       nextToSources ||
       (fs.existsSync(outputPath) && fs.statSync(outputPath).isDirectory())
     ) {
-      if (options.globalTypesFile) {
-        const globalTypesDir = path.dirname(options.globalTypesFile);
-        if (!fs.existsSync(globalTypesDir)) {
-          fs.mkdirSync(globalTypesDir);
-        }
-      } else if (nextToSources && !fs.existsSync(outputPath)) {
-        fs.mkdirSync(outputPath);
-      }
-
       const globalSourcePath =
         options.globalTypesFile ||
         path.join(
           outputPath,
           `globalTypes.${options.tsFileExtension ||
             TYPESCRIPT_DEFAULT_FILE_EXTENSION}`
         );
-      outFiles[globalSourcePath] = {
-        output: generatedGlobalFile.fileContents
-      };
+
+      if (context.typesUsed.length > 0) {
```

Comment:
`context.typesUsed.filter(t => !isScalarType(t)).length > 0`
to avoid custom scalar types

Reply:
Happy to make this change, though it should probably have a test. What's the behavior you're looking for, exactly?

Reply:
Custom scalars like (in schema terms): `scalar MyCustomScalar` are inside this array of typesUsed. However, these do not generate anything in the globals files since the generator does can not know what the data structure is actually. This would still leading to an empty file.

https://www.apollographql.com/docs/apollo-server/schema/custom-scalars/

The test scenario would require a custom scalar type and a query that does not use any enums or other globals, but returns the custom scalar.

Reply:
Thanks. I added the check in 6d42e81.